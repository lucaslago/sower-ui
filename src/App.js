import injectTapEventPlugin from 'react-tap-event-plugin';
injectTapEventPlugin();
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import AppBar from 'material-ui/AppBar';
import React, { Component } from 'react';
import LoginForm from './login/LoginForm';

class App extends Component {
  render() {
    return (
        <MuiThemeProvider>
          <div>
            <AppBar title="Sower"
              iconClassNameRight="muidocs-icon-navigation-expand-more"
            />
            <LoginForm />
          </div>
        </MuiThemeProvider>
    );
  }
}

export default App;
