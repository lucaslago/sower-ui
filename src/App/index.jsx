import React, { Component } from 'react';
import { withRouter } from 'react-router';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import AppBar from 'material-ui/AppBar';
import FlatButton from 'material-ui/FlatButton';
import { Grid } from 'react-bootstrap';
import baseTheme from '../styles/themes/baseTheme';

const barStyle = {
  position: 'fixed'
};

export class App extends Component {
  constructor(props) {
    super(props);
    this.signOut = this.signOut.bind(this);
  }

  signOut() {
    this.props.route.authService.logout();
    this.props.router.push('login');
  }

  render() {
    let icon;
    if (this.props.route.authService.loggedIn()) {
      icon = <FlatButton label="sign out" onClick={this.signOut} />;
    }

    return (
      <MuiThemeProvider muiTheme={baseTheme}>
        <div>
          <AppBar
            title="Sower"
            iconElementRight={icon}
            style={barStyle}
          />
          <Grid>
            { this.props.children }
          </Grid>
        </div>
      </MuiThemeProvider>
    );
  }
}

App.propTypes = {
  children: React.PropTypes.element.isRequired,
  router: React.PropTypes.shape({
    push: React.PropTypes.func.isRequired,
  }),
  route: React.PropTypes.shape({
    authService: React.PropTypes.shape({
      logout: React.PropTypes.func.isRequired,
      loggedIn: React.PropTypes.func.isRequired,
    }),
  }),
};

export default withRouter(App);
