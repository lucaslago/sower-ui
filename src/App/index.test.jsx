import App from './index';

describe('<App/>', () => {
  it('renders without crashing', () => {
    const wrapper = mount(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
