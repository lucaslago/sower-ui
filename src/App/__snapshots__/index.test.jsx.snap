exports[`<App/> renders without crashing 1`] = `
<MuiThemeProvider
  muiTheme={
    Object {
      "appBar": Object {
        "color": "#AE132A",
        "height": 64,
        "padding": 24,
        "textColor": "#ffffff",
        "titleFontWeight": 400,
      },
      "avatar": Object {
        "backgroundColor": "rgb(188, 188, 188)",
        "color": "#ffffff",
      },
      "badge": Object {
        "color": "#ffffff",
        "fontWeight": 500,
        "primaryColor": "#AE132A",
        "primaryTextColor": "#ffffff",
        "secondaryColor": "#ff4081",
        "secondaryTextColor": "#ffffff",
        "textColor": "rgba(0, 0, 0, 0.87)",
      },
      "baseTheme": Object {
        "fontFamily": "Roboto, sans-serif",
        "palette": Object {
          "accent1Color": "#ff4081",
          "accent2Color": "#f5f5f5",
          "accent3Color": "#9e9e9e",
          "alternateTextColor": "#ffffff",
          "borderColor": "#e0e0e0",
          "canvasColor": "#ffffff",
          "clockCircleColor": "rgba(0, 0, 0, 0.07)",
          "disabledColor": "rgba(0, 0, 0, 0.3)",
          "pickerHeaderColor": "#00bcd4",
          "primary1Color": "#AE132A",
          "primary2Color": "#0097a7",
          "primary3Color": "#bdbdbd",
          "secondaryTextColor": "rgba(0, 0, 0, 0.54)",
          "shadowColor": "rgba(0, 0, 0, 1)",
          "textColor": "rgba(0, 0, 0, 0.87)",
        },
        "spacing": Object {
          "desktopDrawerMenuItemHeight": 48,
          "desktopDropDownMenuFontSize": 15,
          "desktopDropDownMenuItemHeight": 32,
          "desktopGutter": 24,
          "desktopGutterLess": 16,
          "desktopGutterMini": 8,
          "desktopGutterMore": 32,
          "desktopKeylineIncrement": 64,
          "desktopSubheaderHeight": 48,
          "desktopToolbarHeight": 56,
          "iconSize": 24,
        },
      },
      "bottomNavigation": Object {
        "backgroundColor": "#ffffff",
        "height": 56,
        "selectedColor": "#AE132A",
        "selectedFontSize": 14,
        "unselectedColor": "rgba(0, 0, 0, 0.54)",
        "unselectedFontSize": 12,
      },
      "button": Object {
        "height": 36,
        "iconButtonSize": 48,
        "minWidth": 88,
      },
      "card": Object {
        "fontWeight": 500,
        "subtitleColor": "rgba(0, 0, 0, 0.54)",
        "titleColor": "rgba(0, 0, 0, 0.87)",
      },
      "cardMedia": Object {
        "color": "rgba(255, 255, 255, 0.87)",
        "overlayContentBackground": "rgba(0, 0, 0, 0.54)",
        "subtitleColor": "rgba(255, 255, 255, 0.54)",
        "titleColor": "rgba(255, 255, 255, 0.87)",
      },
      "cardText": Object {
        "textColor": "rgba(0, 0, 0, 0.87)",
      },
      "checkbox": Object {
        "boxColor": "rgba(0, 0, 0, 0.87)",
        "checkedColor": "#AE132A",
        "disabledColor": "rgba(0, 0, 0, 0.3)",
        "labelColor": "rgba(0, 0, 0, 0.87)",
        "labelDisabledColor": "rgba(0, 0, 0, 0.3)",
        "requiredColor": "#AE132A",
      },
      "chip": Object {
        "backgroundColor": "rgb(224, 224, 224)",
        "deleteIconColor": "rgba(0, 0, 0, 0.26)",
        "fontSize": 14,
        "fontWeight": 400,
        "shadow": "0 1px 6px rgba(0, 0, 0, 0.12),
            0 1px 4px rgba(0, 0, 0, 0.12)",
        "textColor": "rgba(0, 0, 0, 0.87)",
      },
      "datePicker": Object {
        "calendarTextColor": "rgba(0, 0, 0, 0.87)",
        "calendarYearBackgroundColor": "#ffffff",
        "color": "#AE132A",
        "selectColor": "#0097a7",
        "selectTextColor": "#ffffff",
        "textColor": "#ffffff",
      },
      "dialog": Object {
        "bodyColor": "rgba(0, 0, 0, 0.6)",
        "bodyFontSize": 16,
        "titleFontSize": 22,
      },
      "drawer": Object {
        "color": "#ffffff",
        "width": 256,
      },
      "dropDownMenu": Object {
        "accentColor": "#e0e0e0",
      },
      "enhancedButton": Object {
        "tapHighlightColor": "rgba(0, 0, 0, 0)",
      },
      "flatButton": Object {
        "buttonFilterColor": "#999999",
        "color": "rgba(0, 0, 0, 0)",
        "disabledTextColor": "rgba(0, 0, 0, 0.3)",
        "fontSize": 14,
        "fontWeight": 500,
        "primaryTextColor": "#AE132A",
        "secondaryTextColor": "#ff4081",
        "textColor": "rgba(0, 0, 0, 0.87)",
      },
      "floatingActionButton": Object {
        "buttonSize": 56,
        "color": "#AE132A",
        "disabledColor": "rgb(224, 224, 224)",
        "disabledTextColor": "rgba(0, 0, 0, 0.3)",
        "iconColor": "#ffffff",
        "miniSize": 40,
        "secondaryColor": "#ff4081",
        "secondaryIconColor": "#ffffff",
      },
      "fontFamily": "Roboto, sans-serif",
      "gridTile": Object {
        "textColor": "#ffffff",
      },
      "icon": Object {
        "backgroundColor": "#AE132A",
        "color": "#ffffff",
      },
      "inkBar": Object {
        "backgroundColor": "#ff4081",
      },
      "isRtl": false,
      "listItem": Object {
        "leftIconColor": "#757575",
        "nestedLevelDepth": 18,
        "rightIconColor": "#757575",
        "secondaryTextColor": "rgba(0, 0, 0, 0.54)",
      },
      "menu": Object {
        "backgroundColor": "#ffffff",
        "containerBackgroundColor": "#ffffff",
      },
      "menuItem": Object {
        "dataHeight": 32,
        "height": 48,
        "hoverColor": "rgba(0, 0, 0, 0.1)",
        "padding": 24,
        "rightIconDesktopFill": "#757575",
        "selectedTextColor": "#ff4081",
      },
      "menuSubheader": Object {
        "borderColor": "#e0e0e0",
        "padding": 24,
        "textColor": "#AE132A",
      },
      "overlay": Object {
        "backgroundColor": "rgba(0, 0, 0, 0.54)",
      },
      "palette": Object {
        "accent1Color": "#ff4081",
        "accent2Color": "#f5f5f5",
        "accent3Color": "#9e9e9e",
        "alternateTextColor": "#ffffff",
        "borderColor": "#e0e0e0",
        "canvasColor": "#ffffff",
        "clockCircleColor": "rgba(0, 0, 0, 0.07)",
        "disabledColor": "rgba(0, 0, 0, 0.3)",
        "pickerHeaderColor": "#00bcd4",
        "primary1Color": "#AE132A",
        "primary2Color": "#0097a7",
        "primary3Color": "#bdbdbd",
        "secondaryTextColor": "rgba(0, 0, 0, 0.54)",
        "shadowColor": "rgba(0, 0, 0, 1)",
        "textColor": "rgba(0, 0, 0, 0.87)",
      },
      "paper": Object {
        "backgroundColor": "#ffffff",
        "color": "rgba(0, 0, 0, 0.87)",
        "zDepthShadows": Array [
          "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
          "0 3px 10px rgba(0, 0, 0, 0.16),
             0 3px 10px rgba(0, 0, 0, 0.23)",
          "0 10px 30px rgba(0, 0, 0, 0.19),
             0 6px 10px rgba(0, 0, 0, 0.23)",
          "0 14px 45px rgba(0, 0, 0, 0.25),
             0 10px 18px rgba(0, 0, 0, 0.22)",
          "0 19px 60px rgba(0, 0, 0, 0.3),
             0 15px 20px rgba(0, 0, 0, 0.22)",
        ],
      },
      "prepareStyles": [Function],
      "radioButton": Object {
        "backgroundColor": "#ffffff",
        "borderColor": "rgba(0, 0, 0, 0.87)",
        "checkedColor": "#AE132A",
        "disabledColor": "rgba(0, 0, 0, 0.3)",
        "labelColor": "rgba(0, 0, 0, 0.87)",
        "labelDisabledColor": "rgba(0, 0, 0, 0.3)",
        "requiredColor": "#AE132A",
        "size": 24,
      },
      "raisedButton": Object {
        "color": "#ffffff",
        "disabledColor": "rgb(229, 229, 229)",
        "disabledTextColor": "rgba(0, 0, 0, 0.3)",
        "fontSize": 14,
        "fontWeight": 500,
        "primaryColor": "#AE132A",
        "primaryTextColor": "#ffffff",
        "secondaryColor": "#ff4081",
        "secondaryTextColor": "#ffffff",
        "textColor": "rgba(0, 0, 0, 0.87)",
      },
      "rawTheme": Object {
        "fontFamily": "Roboto, sans-serif",
        "palette": Object {
          "accent1Color": "#ff4081",
          "accent2Color": "#f5f5f5",
          "accent3Color": "#9e9e9e",
          "alternateTextColor": "#ffffff",
          "borderColor": "#e0e0e0",
          "canvasColor": "#ffffff",
          "clockCircleColor": "rgba(0, 0, 0, 0.07)",
          "disabledColor": "rgba(0, 0, 0, 0.3)",
          "pickerHeaderColor": "#00bcd4",
          "primary1Color": "#AE132A",
          "primary2Color": "#0097a7",
          "primary3Color": "#bdbdbd",
          "secondaryTextColor": "rgba(0, 0, 0, 0.54)",
          "shadowColor": "rgba(0, 0, 0, 1)",
          "textColor": "rgba(0, 0, 0, 0.87)",
        },
        "spacing": Object {
          "desktopDrawerMenuItemHeight": 48,
          "desktopDropDownMenuFontSize": 15,
          "desktopDropDownMenuItemHeight": 32,
          "desktopGutter": 24,
          "desktopGutterLess": 16,
          "desktopGutterMini": 8,
          "desktopGutterMore": 32,
          "desktopKeylineIncrement": 64,
          "desktopSubheaderHeight": 48,
          "desktopToolbarHeight": 56,
          "iconSize": 24,
        },
      },
      "refreshIndicator": Object {
        "loadingStrokeColor": "#AE132A",
        "strokeColor": "#e0e0e0",
      },
      "ripple": Object {
        "color": "rgba(0, 0, 0, 0.87)",
      },
      "slider": Object {
        "handleColorZero": "#bdbdbd",
        "handleFillColor": "#ffffff",
        "handleSize": 12,
        "handleSizeActive": 18,
        "handleSizeDisabled": 8,
        "rippleColor": "#AE132A",
        "selectionColor": "#AE132A",
        "trackColor": "#bdbdbd",
        "trackColorSelected": "#9e9e9e",
        "trackSize": 2,
      },
      "snackbar": Object {
        "actionColor": "#ff4081",
        "backgroundColor": "rgba(0, 0, 0, 0.87)",
        "textColor": "#ffffff",
      },
      "spacing": Object {
        "desktopDrawerMenuItemHeight": 48,
        "desktopDropDownMenuFontSize": 15,
        "desktopDropDownMenuItemHeight": 32,
        "desktopGutter": 24,
        "desktopGutterLess": 16,
        "desktopGutterMini": 8,
        "desktopGutterMore": 32,
        "desktopKeylineIncrement": 64,
        "desktopSubheaderHeight": 48,
        "desktopToolbarHeight": 56,
        "iconSize": 24,
      },
      "stepper": Object {
        "backgroundColor": "transparent",
        "connectorLineColor": "#bdbdbd",
        "disabledTextColor": "rgba(0, 0, 0, 0.26)",
        "hoverBackgroundColor": "rgba(0, 0, 0, 0.06)",
        "hoveredIconColor": "#616161",
        "iconColor": "#AE132A",
        "inactiveIconColor": "#9e9e9e",
        "textColor": "rgba(0, 0, 0, 0.87)",
      },
      "subheader": Object {
        "color": "rgba(0, 0, 0, 0.54)",
        "fontWeight": 500,
      },
      "svgIcon": Object {
        "color": "rgba(0, 0, 0, 0.87)",
      },
      "table": Object {
        "backgroundColor": "#ffffff",
      },
      "tableFooter": Object {
        "borderColor": "#e0e0e0",
        "textColor": "#9e9e9e",
      },
      "tableHeader": Object {
        "borderColor": "#e0e0e0",
      },
      "tableHeaderColumn": Object {
        "height": 56,
        "spacing": 24,
        "textColor": "#9e9e9e",
      },
      "tableRow": Object {
        "borderColor": "#e0e0e0",
        "height": 48,
        "hoverColor": "#f5f5f5",
        "selectedColor": "#e0e0e0",
        "stripeColor": "rgba(214, 137, 148, 0.4)",
        "textColor": "rgba(0, 0, 0, 0.87)",
      },
      "tableRowColumn": Object {
        "height": 48,
        "spacing": 24,
      },
      "tabs": Object {
        "backgroundColor": "#AE132A",
        "selectedTextColor": "#ffffff",
        "textColor": "rgba(255, 255, 255, 0.7)",
      },
      "textField": Object {
        "backgroundColor": "transparent",
        "borderColor": "#e0e0e0",
        "disabledTextColor": "rgba(0, 0, 0, 0.3)",
        "errorColor": "#f44336",
        "floatingLabelColor": "rgba(0, 0, 0, 0.3)",
        "focusColor": "#AE132A",
        "hintColor": "rgba(0, 0, 0, 0.3)",
        "textColor": "rgba(0, 0, 0, 0.87)",
      },
      "timePicker": Object {
        "accentColor": "#AE132A",
        "clockCircleColor": "rgba(0, 0, 0, 0.07)",
        "clockColor": "rgba(0, 0, 0, 0.87)",
        "color": "#ffffff",
        "headerColor": "#00bcd4",
        "selectColor": "#0097a7",
        "selectTextColor": "#ffffff",
        "textColor": "#9e9e9e",
      },
      "toggle": Object {
        "labelColor": "rgba(0, 0, 0, 0.87)",
        "labelDisabledColor": "rgba(0, 0, 0, 0.3)",
        "thumbDisabledColor": "#e0e0e0",
        "thumbOffColor": "#f5f5f5",
        "thumbOnColor": "#AE132A",
        "thumbRequiredColor": "#AE132A",
        "trackDisabledColor": "#bdbdbd",
        "trackOffColor": "#bdbdbd",
        "trackOnColor": "rgba(174, 19, 42, 0.5)",
        "trackRequiredColor": "rgba(174, 19, 42, 0.5)",
      },
      "toolbar": Object {
        "backgroundColor": "rgb(232, 232, 232)",
        "color": "rgba(0, 0, 0, 0.54)",
        "height": 56,
        "hoverColor": "rgba(0, 0, 0, 0.87)",
        "iconColor": "rgba(0, 0, 0, 0.4)",
        "menuHoverColor": "rgba(0, 0, 0, 0.1)",
        "separatorColor": "rgba(0, 0, 0, 0.175)",
        "titleFontSize": 20,
      },
      "tooltip": Object {
        "color": "#ffffff",
        "rippleBackgroundColor": "#616161",
      },
      "zIndex": Object {
        "appBar": 1100,
        "dialog": 1500,
        "dialogOverlay": 1400,
        "drawer": 1300,
        "drawerOverlay": 1200,
        "layer": 2000,
        "menu": 1000,
        "popover": 2100,
        "snackbar": 2900,
        "tooltip": 3000,
      },
    }
  }>
  <div>
    <AppBar
      showMenuIconButton={true}
      style={
        Object {
          "position": "fixed",
        }
      }
      title="Sower"
      zDepth={1} />
    <Grid
      bsClass="container"
      componentClass="div"
      fluid={false}>
      <div>
        <span>
           mock 
        </span>
        <span>
           component 
        </span>
      </div>
    </Grid>
  </div>
</MuiThemeProvider>
`;
